File: leetcode_java/27. Remove Element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int i = 0, j = nums.length - 1;
        while(i <= j) {
            if(nums[i] == val) {
                if(nums[j] != val) nums[i] = nums[j];
                j--;
            }
            else i++;
        }
        return j + 1;
    }
}

File: leetcode_python/27. Remove Element.py
class Solution:
    def removeElement(self, nums: [int], val: int) -> int:
        i, j = 0, len(nums) - 1
        while i <= j:
            if nums[i] == val:
                if nums[j] != val: nums[i] = nums[j]
                j -= 1
            else: i += 1
        return j + 1